game-core/src/test/java/games/strategy/triplea/image/UnitImageFactoryTest.java

```diff
@@ -0,0 +1,393 @@
+package games.strategy.triplea.image;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.lenient;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.MutableProperty;
+import games.strategy.engine.data.TechnologyFrontier;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class UnitImageFactoryTest {
+  @Mock GameData gameData;
+
+  @Nested
+  class GetBaseImageName {
+
+    @BeforeEach
+    void setupGameData() {
+      final TechnologyFrontier technologyFrontier = new TechnologyFrontier("Tech", gameData);
+      when(gameData.getTechnologyFrontier()).thenReturn(technologyFrontier);
+      lenient().when(gameData.getDiceSides()).thenReturn(6);
+      final GameProperties gameProperties = new GameProperties(gameData);
+      lenient().when(gameData.getProperties()).thenReturn(gameProperties);
+    }
+
+    @Test
+    void basicUnitType() throws MutableProperty.InvalidValueException {
+      givenUnitTypeAndTechnologyAssertThatImageHasSuffix(List.of(), List.of(), "");
+    }
+
+    @Test
+    void basicDamagedUnitType() throws MutableProperty.InvalidValueException {
+      givenUnitTypeAndTechnologyAssertThatImageHasName(
+          "infantry", List.of(), List.of(), true, false, "infantry_hit");
+    }
+
+    @Test
+    void basicDisabledUnitType() throws MutableProperty.InvalidValueException {
+      givenUnitTypeAndTechnologyAssertThatImageHasName(
+          "infantry", List.of(), List.of(), false, true, "infantry_disabled");
```

Comment:
Boolean parameters can be difficult to understand. Overloaded methods or a parameter object can help clarify this.

For example, with a parameter object, we could have something like:
```


givenTestSetup(SetupParameters.builder()
       .unitTypeName("infantry")
        .unitAttachmentProperties(List.of())
        .techAttachmentProperties(List.of())
        .damaged(false)
        .disabled(true))
   .build());

final String imageName =
          UnitImageFactory.getBaseImageName(unitType, player, damaged, disabled);

assertThat(imageName, is("infantry_disabled");
```

The above refactor of course does not work since we need direct parameters.. This indicates a production code change to accept a parameter object is perhaps all we need and then the testing becomes drastically simplified.

To get there, you can create a method overload that accepts the parameter object and then invokes the original method. Then you can invoke the method overload from test. Once you have the test in place, it's a question of migrating the rest of the production code to use the better API.

EG:

```
final String imageName =
          UnitImageFactory.getBaseImageName(  ImageNameParameters.builder()
                .unitTypeName("infantry")
                .unitAttachmentProperties(List.of())
                .techAttachmentProperties(List.of())
                .damaged(false)
                .disabled(true))
   .build());

assertThat(imageName, is("infantry_disabled");
```

From here, test parameterization could help and the API's would be straight forward. EG:

@ParameterizedTest
@MethodSource
void verifyBaseImageName(ImageNameParameters. parameters, String expectedName) {
    final String imageName =
            UnitImageFactory.getBaseImageName(parameters);

     assertThat(imageName, is(expectedName);
}
```


Thoughts @trevan ?